libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,300 @@
+package protocols
+
+type Address string
+type Performative []string
+type SomeMessageType string
+
+const (
+	NONCE_BYTES_NB = 32
+)
+
+// type Dialogue struct {
+// 	_dialogues_storage
+// 	_dialogues_stats
+// 	_keep_terminal_state_dialogues
+// 	_message_class
+// 	_dialogue_class
+// 	_role_from_first_message
+// 	_self_address
+// 	_outgoing_messages
+// 	_incoming_messages
+// }
+
+type Role string
+
+const (
+	Role1 Role = "role1"
+	Role2 Role = "role2"
```

Comment:
these are protocol specific, so not defined here

Reply:
Okay, could I keep the type as a string that would act as the unique identifier for the specific protocol? Will this be passed with a message when initializing the dialogue?